sources/packages/backend/libs/sims-db/src/entities/education-program.model.ts

```diff
@@ -357,6 +361,29 @@ export class EducationProgram extends RecordDataModel {
   })
   effectiveEndDate?: string;
 
+  /**
+   * Indicates whether an education program has reached its effective end date or not.
+   * Note that:
+   *   - It is not updated for typeOrm raw results queries (i.e. getRawOne and getRawMany).
+   *   - `effectiveEndDate` should have been selected in the query for this to be properly evaluated.
+   */
+  isExpired: boolean;
+
+  /**
+   * Updates `isExpired` property with `true` in case the effective end date has been reached or
+   * the effective end date is null. Otherwise `false`.
+   * This method is called after retrieve, insert or update operations.
+   */
+  @AfterLoad()
+  @AfterInsert()
+  @AfterUpdate()
+  updateIsExpired() {
+    if (!this.effectiveEndDate) this.isExpired = false;
+    else
+      this.isExpired =
+        getISODateOnlyString(new Date()) >= this.effectiveEndDate;
+  }
```

Comment:
To make it centralized and simple, unless there is a performance impact, we adopted the solution as below. In the below way we make it clear that it is a read-only property, works whatever the dependent properties are present, and keeps it centralized.
```ts
/**
   * Is the institution type BC Private.
   */
  get isBCPrivate(): boolean {
    return INSTITUTION_TYPE_BC_PRIVATE === this.id;
  }
```
In this case it would be.
```ts
 get isExpired(): boolean {
    return !this.effectiveEndDate || new Date() >= new Date(this.effectiveEndDate);
  }
```